### PR TITLE
Add workspace icons

### LINKED ISSUE #6204: Redesign: Workspace iconography
## Feature Description

Update iconography for workspace buttons based on [Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories---Post-Stable?node-id=1009%3A4323).

Pay attention to both sizing and spacing.

- [x] Update next/previous arrows for both page and carousel
- [x] Update page menu icons (delete, duplicate, add, undo, redo)
- [x] Update carousel menu icons (keyboard, gridview)

For this ticket: Ignore version number, ignore zoom dropdown.

| Current design | New design |
|-|-|
| Screen Shot 2021-02-04 at 22 39 46 | Screen Shot 2021-02-04 at 22 37 49 |

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
## Context

See #6204 for todo.

### QA

This PR can be tested by following these steps:

1. Load up a story
2. Observe the new icons in the page navigation, page menu, carousel, and carousel menu
3. Verify that all the above-mentioned icons work as intended (as before)

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6204

### CODE REVIEW

File: assets/src/edit-story/theme.js
Comment (barklund) [inline assets/src/edit-story/theme.js]: Note that this is a temporary workaround until #6209 solves this correctly.

File: assets/src/edit-story/theme.js
Comment (barklund) [inline assets/src/edit-story/theme.js]: We have this through some other dependency, so it builds, but we don't directly depend on it, so it doesn't lint.

**Comment by barklund** [comment]
@miina, I have many open questions about this still, and it's blocked by #6209, but please take a look anyway. I like this new style of buttons fairly well - and the result looks great both in code and editor!

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [inline assets/src/edit-story/components/canvas/pagemenu/index.js:290]: I decided to fully remove this (but not the underlying flag) as it was just some old icons lying around with no accompanying functionality. If we ever need this functionality, design has to be redone anyway.

@swissspidy, should I remove the flag as well?

File: assets/src/edit-story/components/canvas/layout.js
Comment (barklund) [inline assets/src/edit-story/components/canvas/layout.js]: This (random magic number) will all be done [properly](https://github.com/google/web-stories-wp/pull/6236/files#diff-f634fcb15355d7101d454a9ad3b2aa06ceb1ce6bd5ebd0f4e8dcf55c9ded3441R59) in the zoom PR in #6236 , so ignore for this one.

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [inline assets/src/edit-story/components/canvas/pagemenu/index.js]: So the icons happen to be half the height of the buttons and the divider should follow the icon height... Is there a cleaner way to do this?

File: assets/src/edit-story/theme.js
Comment (swissspidy) [reply assets/src/edit-story/theme.js]: It would need to be added to `package.json` explicitly

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (miina) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: Maybe an option could be adding the sizes of the button variants in pixels as well to the `button/index.js` in the design-system folder and using that instead of height? Would be great if we could still access the size in px somehow as well for these cases specifically -- when other things depend on the size, might not be the only time when this happens. Then, inside the `button/index.js` also these constants could be used throughout the file instead of using numbers directly. Thoughts?

File: assets/src/edit-story/components/canvas/layout.js
Comment (miina) [inline assets/src/edit-story/components/canvas/layout.js]: Hm, what's the overflow for?

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (swissspidy) [reply assets/src/edit-story/components/canvas/pagemenu/index.js:290]: Yeah let's also remove the flag.

File: assets/src/edit-story/components/carousel/carouselScroll.js
Comment (miina) [inline assets/src/edit-story/components/carousel/carouselScroll.js]: Isn't this the default?

File: assets/src/edit-story/components/carousel/carouselLayout.js
Comment (miina) [inline assets/src/edit-story/components/carousel/carouselLayout.js:30]: Here, too -- there are two different general button sizes (not only related to the carousel) that seem to be needed elsewhere, too, so this should probably come from the design system (`buttons/index.js` or separate constants file) instead.

File: assets/src/edit-story/components/canvas/pagenav/index.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagenav/index.js:62]: Tried to test on `main` to understand what was happening before but wasn't sure. Curious what was the previous behavior that this prevents now for the user.

File: assets/src/edit-story/components/canvas/pagenav/index.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagenav/index.js]: Here, too, isn't this the default?

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: So, I added the pixel sizes of icon buttons to the button constants in the design system. I'm not loving it, but it works okay. Naming could be optimized.

I'd like input from @BrittanyIRL as well here.

* Adding constants: dda29454e
* Using them: 0fa5c1cc4

Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: (also, the icons weren't actually half the size - but slightly smaller than that even: 14px vs 32px button)

File: assets/src/edit-story/components/canvas/layout.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/layout.js]: The focus border overflows. Without it you get:

Screen Shot 2021-02-09 at 17 22 01

Which looks stupid 🤷 . And I don't think the overflow here hurts.

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (BrittanyIRL) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: I feel ok about this because we're already defining the height in pixels here. I would say 1 nit would be can we name the  group `BUTTON_HEIGHT_PIXELS` or something similar that can give us a bit of context as to what the pixels are? 

My assumption reading through this is that the divider would always want to be relative to the button sizes so it makes sense to tie it together.

File: assets/src/edit-story/components/carousel/carouselLayout.js
Comment (barklund) [reply assets/src/edit-story/components/carousel/carouselLayout.js:30]: Referring to the [previous discussion](https://github.com/google/web-stories-wp/pull/6272#discussion_r573286458).

File: assets/src/edit-story/components/canvas/pagenav/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagenav/index.js:62]: It was also there [before](https://github.com/google/web-stories-wp/pull/6272/files/ac66669804e66fce95eeda6eddbfe9059ee26739#diff-02139befcc819b7d91d045078dd199c489e48c9dce377d439faf7b0dbcdc8c52L90) - I just memoized it.

File: assets/src/edit-story/components/carousel/carouselScroll.js
Comment (barklund) [reply assets/src/edit-story/components/carousel/carouselScroll.js]: Indeed! Fixed in 0000c9331 .

File: assets/src/edit-story/components/canvas/pagenav/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagenav/index.js]: Fixed in 0000c9331.

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: Well, it is both the width and the height of the elements (and in other places we're using it to express width), so that's why the dimension-neutral name. And `SIZE` was already taken.

But it could perhaps be `BUTTON_DIMENSIONS` instead?

File: assets/src/edit-story/theme.js
Comment (barklund) [reply assets/src/edit-story/theme.js]: It's temporary only until #6209 is merged in, that's why I didn't add it. I just needed it to test locally.

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js:290]: Removed in 0a21ace.

**Comment by barklund** [comment]
Currently awaiting discussion on icon sizing before adding the new correct icons required.

File: assets/src/edit-story/theme.js
Comment (barklund) [reply assets/src/edit-story/theme.js]: Now removed

File: assets/src/edit-story/theme.js
Comment (barklund) [reply assets/src/edit-story/theme.js]: Now removed again.

**Comment by barklund** [comment]
> @miina, I have many open questions about this still, and it's blocked by #6209, but please take a look anyway. I like this new style of buttons fairly well - and the result looks great both in code and editor!

And it's now ready for a full review, @miina.

File: assets/src/edit-story/components/canvas/pagemenu/index.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: Changed all this completely, as icons only have one size now. There's a theme constant for the large button size and another for the icon size. That's it.

Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/index.js]: (and neither is used for this particular variable - it's completely custom set to 16px, which looks good, but doesn't directly match the icon heights, as they vary a slight bit)

File: assets/src/design-system/icons/chevron_up_small.svg
Comment (swissspidy) [inline assets/src/design-system/icons/chevron_up_small.svg]: FWIW our SVGO config should be doing this automatically under the hood.

Good practice nontheless 👍

File: assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js:35]: Use `Tooltip` from the `design-system` instead?

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagemenu/pageMenu.js]: Nit: Should `left` and `right` be with capital letters, too? `ArrowDownLeftCurved` etc.

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagemenu/pageMenu.js:102]: 280 to a constant perhaps?

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (miina) [inline assets/src/edit-story/components/canvas/pagemenu/pageMenu.js]: Should these be moved to the design-system under `typography`, too?

File: assets/src/edit-story/components/canvas/layout.js
Comment (miina) [inline assets/src/edit-story/components/canvas/layout.js:89]: Yay! Almost looks like a room full of (lined up) monsters in [Nethack](https://en.wikipedia.org/wiki/NetHack) 😍 .

File: assets/src/design-system/icons/chevron_up_small.svg
Comment (barklund) [reply assets/src/design-system/icons/chevron_up_small.svg]: Might this be missing for Storybook only then? Because without it, the icons clearly weren't colorable (in Storybook)) - which is why I noticed these exact problems.

Comment (swissspidy) [reply assets/src/design-system/icons/chevron_up_small.svg]: Strange.

We definitely do load svgr in our storybook webpack config:

https://github.com/google/web-stories-wp/blob/38e3513c7e8ba1d46c5bde5d5572c662d89803cc/.storybook/main.cjs#L52

If that's broken, would you mind opening a ticket for us to investigate? Thanks!

Comment (barklund) [reply assets/src/design-system/icons/chevron_up_small.svg]: I can confirm it doesn't work in Storybook, only in the actual editor build. Maybe it's because `svgo.yml` is in the root directory, so it's not picked up inside the `storybook` folder? Follow-up ticket in #6357.

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/pageMenu.js]: I could do [even better actually](https://github.com/google/web-stories-wp/pull/6272/files#diff-e12246607f5678a452584b8ec9cfc1990d830a3db193f87d7bf2b385ed760b0dR114-R120)! Thanks for pushing me when I'm lazy 😉 .

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/pageMenu.js:102]: It's temporary only until zoom is merged in - then this will change to some actual calculated value.

File: assets/src/edit-story/components/canvas/pagemenu/pageMenu.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/pageMenu.js]: Done in [the icon PR](https://github.com/google/web-stories-wp/pull/6311/commits/4b24d3ce8c32d41a1c6daac2ef80bd024c8ea67d).

File: assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js
Comment (barklund) [reply assets/src/edit-story/components/canvas/pagemenu/pageMenuButton.js:35]: Yes, we should implement that too. But let's hold off on doing it piecemeal and implement it through somewhere. It's probably easiest to change the insides of `WithTooltip` instead actually. Noted for future ticket 📝

**Review by miina — APPROVED** [review]
LGTM Pending the icons PR merge

**Pull request merged**
